iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -10,28 +10,21 @@ import UIKit
 protocol TimeMachineLayoutItemType {
     /// A unique identifier for the item.
     var layoutId: String { get }
+    /// A date associated with the time machine item
+    var date: Date { get }
 }
 
 protocol TimeMachineCollectionViewLayoutDataSource: AnyObject {
     func getTimeMachineItem(forItemAt indexPath: IndexPath) -> TimeMachineLayoutItemType
 }
 
-class TimeMachineCollectionViewLayoutInvalidationContext: UICollectionViewLayoutInvalidationContext {
-    /// If true, the z ranges for all the items should be recalculated.
-    var shouldRecalculateZRanges = true
-}
-
 /// A custom layout for a stack of cells laid out along the z axis.
 /// The stacks appear similar to Apple's Time Machine interface, with the newest item in front and older items going out into the distance.
 /// As the collection view scrolls up and down, the items move away and toward the user respectively.
 class TimeMachineCollectionViewLayout: UICollectionViewLayout {
 
     typealias SectionIndex = Int
 
-    override class var invalidationContextClass: AnyClass {
-        return TimeMachineCollectionViewLayoutInvalidationContext.self
-    }
-
```

Comment:
The custom invalidation class is no longer needed now that there's only one stack.